Rivet/Functions/Convert-FileInfoToMigration.ps1

```diff
@@ -34,63 +34,63 @@ function Convert-FileInfoToMigration
-                Write-Timing -Message ('Convert-FileInfoToMigration  {0}' -f $m.FullName)
-
-                filter Add-Operation
-                {
-                    param(
-                        [Parameter(Mandatory=$true,ValueFromPipeline=$true)]
-                        [object]
-                        # The migration object to invoke.
-                        $Operation,
-
-                        [Parameter(ParameterSetName='Push',Mandatory=$true)]
-                        [Collections.Generic.List[Rivet.Operations.Operation]]
-                        [AllowEmptyCollection()]
-                        $OperationsList,
+        filter Add-Operation
+        {
+            param(
+                [Parameter(Mandatory,ValueFromPipeline)]
+                # The migration object to invoke.
+                [Object]$Operation,
 
-                        [Parameter(ParameterSetName='Pop',Mandatory=$true)]
-                        [Switch]
-                        $Pop
-                    )
+                [Parameter(ParameterSetName='Push',Mandatory)]
+                [AllowEmptyCollection()]
+                [Collections.Generic.List[Rivet.Operations.Operation]]$OperationsList,
 
-                    Set-StrictMode -Version 'Latest'
+                [Parameter(ParameterSetName='Pop',Mandatory)]
+                [switch]$Pop
+            )
 
+            Set-StrictMode -Version 'Latest'
 
-                    $Operation |
-                        Where-Object { $_ -is [Rivet.Operations.Operation] } |
-                        ForEach-Object {
+            foreach( $operationItem in $Operation )
+            {
+                if( $operationItem -isnot [Rivet.Operations.Operation] )
+                {
+                    continue
+                }
 
-                            $pluginParameter = @{ Migration = $m ; Operation = $_ }
+                $pluginParameter = @{ Migration = $m ; Operation = $_ }
 
-                            Invoke-RivetPlugin -Event ([Rivet.Events]::BeforeOperationLoad) -Parameter $pluginParameter
+                [Rivet.Operations.Operation[]]$operations = & {
+                        Invoke-RivetPlugin -Event ([Rivet.Events]::BeforeOperationLoad) -Parameter $pluginParameter
+                        $operationItem
+                        Invoke-RivetPlugin -Event ([Rivet.Events]::AfterOperationLoad) -Parameter $pluginParameter
+                    } | 
+                    Where-Object { $_ -is [Rivet.Operations.Operation] } |
+                    Repair-Operation
 
-                            $_
+                $OperationsList.AddRange($operations)
+            }
+        }
 
-                            Invoke-RivetPlugin -Event ([Rivet.Events]::AfterOperationLoad) -Parameter $pluginParameter
+        foreach( $fileInfo in $InputObject )
+        {
+            $dbName = Split-Path -Parent -Path $fileInfo.FullName

```

Comment:
Comment above for intent would make this clearer to me. The multiple re-assignment from similar operations..